pkg/registry/generic/rest/response_checker.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rest
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+)
+
+// Check the http error status from a location URL.
+// And convert an error into a structured API object.
+// Finally ensure we close the body before returning the error
+type HttpResponseChecker interface {
+	Check(resp *http.Response) error
+}
+
+// Max length read from the response body of a location which returns error status
+const (
+	maxReadLength = 50000
+)
+
+// A generic http response checker to transform the error.
+type GenericHttpResponseChecker struct {
+	Kind string
+	Name string
+}
+
+func (checker GenericHttpResponseChecker) Check(resp *http.Response) error {
```

Comment:
Can you add a few test cases for this method specifically (verify limit, verify error on read returns internal error, verify generic server response)?